repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/AssetPreProcessor.java

```diff
@@ -71,39 +67,44 @@ private void processCreateAsset(AtlasEntity entity, AtlasVertex vertex) throws A
     private void processUpdateAsset(AtlasEntity entity, AtlasVertex vertex) throws AtlasBaseException {
         AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processUpdateAsset");
 
-        if(entity.hasAttribute(DOMAIN_GUIDS)) {
-            validateDomainAssetLinks(entity);
-            AtlasEntityHeader sourceEntity = new AtlasEntityHeader(entity);
-            isAuthorized(sourceEntity);
-        }
+        processDomainLinkAttribute(entity);
 
         RequestContext.get().endMetricRecord(metricRecorder);
 
     }
 
+    private void processDomainLinkAttribute(AtlasEntity entity) throws AtlasBaseException {
+        if(entity.hasAttribute(DOMAIN_GUIDS)){
+            validateDomainAssetLinks(entity);
+            isAuthorized(entity);
+        }
+    }
+
     private void validateDomainAssetLinks(AtlasEntity entity) throws AtlasBaseException {
         List<String> domainGuids = ( List<String>) entity.getAttribute(DOMAIN_GUIDS);
 
-        if(domainGuids.size() > 1) {
-            throw new AtlasBaseException(AtlasErrorCode.INVALID_PARAMETERS, "Asset can be linked to only one domain");
-        }
+        if(CollectionUtils.isNotEmpty(domainGuids)){
+            if(domainGuids.size() > 1) {
+                throw new AtlasBaseException(AtlasErrorCode.INVALID_PARAMETERS, "Asset can be linked to only one domain");
+            }
 
-        if(CollectionUtils.isNotEmpty(domainGuids)) {
             for(String domainGuid : domainGuids) {
                 AtlasVertex domainVertex = entityRetriever.getEntityVertex(domainGuid);
-                  if(domainVertex == null) {
-                        throw new AtlasBaseException(AtlasErrorCode.INSTANCE_GUID_NOT_FOUND, domainGuid);
-                  }
-                  else{
-                      LOG.info("Domain vertex found for guid: {}", domainGuid);
-                  }
+                if(domainVertex == null) {
+                    throw new AtlasBaseException(AtlasErrorCode.INSTANCE_GUID_NOT_FOUND, domainGuid);
+                }
+
+                if (!Objects.equals(entity.getTypeName(), DATA_DOMAIN_ENTITY_TYPE)){
```

Comment:
Compare domainVertex's __typeName with DATA_DOMAIN_ENTITY_TYPE

Reply:
Did this change not sure why this is not reflecting